subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskValidationContext.java

```diff
@@ -18,19 +18,28 @@
 
 import org.gradle.api.internal.file.FileResolver;
 import org.gradle.api.internal.tasks.properties.DefaultParameterValidationContext;
+import org.gradle.internal.file.ReservedFileSystemLocationRegistry;
 
+import java.io.File;
 import java.util.Collection;
 
 public class DefaultTaskValidationContext extends DefaultParameterValidationContext implements TaskValidationContext {
     private final FileResolver resolver;
+    private final ReservedFileSystemLocationRegistry reservedFileSystemLocationRegistry;
 
-    public DefaultTaskValidationContext(FileResolver resolver, Collection<String> messages) {
+    public DefaultTaskValidationContext(FileResolver resolver, ReservedFileSystemLocationRegistry reservedFileSystemLocationRegistry, Collection<String> messages) {
         super(messages);
         this.resolver = resolver;
+        this.reservedFileSystemLocationRegistry = reservedFileSystemLocationRegistry;
     }
 
     @Override
     public FileResolver getResolver() {
         return resolver;
     }
+
+    @Override
+    public boolean isInReservedDirectory(File location) {
+        return reservedFileSystemLocationRegistry.isInReservedDirectory(location);
```

Comment:
We should use one term: "directory", "file system location" or "location" are all good as long as we use one.